lighthouse-core/gather/gatherers/seo/tap-targets.js

```diff
@@ -287,17 +352,66 @@ function gatherTapTargets() {
       return;
     }
 
-    targets.push({
-      clientRects: visibleClientRects,
-      snippet: truncate(tapTargetElement.outerHTML, 300),
-      // @ts-ignore - getNodePath put into scope via stringification
-      path: getNodePath(tapTargetElement),
-      // @ts-ignore - getNodeSelector put into scope via stringification
-      selector: getNodeSelector(tapTargetElement),
-      href: /** @type {HTMLAnchorElement} */(tapTargetElement)['href'] || '',
+    const largestRect = getLargestRect(visibleClientRects);
+    const largestRectCenterPoint = getRectCenterPoint(largestRect);
+    // round so we can can assume whole numbers during in-viewport check
+    largestRectCenterPoint.x = Math.round(largestRectCenterPoint.x);
+    largestRectCenterPoint.y = Math.round(largestRectCenterPoint.y);
+
+    if (largestRectCenterPoint.x >= window.innerWidth) {
+      // we don't scroll sideways, so the center of this tap target is always hidden
+      return;
+    }
+
+    enhancedTapTargets.push({
+      tapTargetElement,
+      largestRectCenterPoint,
+      visibleClientRects,
     });
   });
 
+
+  enhancedTapTargets.sort(
+    (a, b) => {
+      return a.largestRectCenterPoint.y - b.largestRectCenterPoint.y;
+    }
+  );
+
+  // Disable point events so that tap targets below them don't get
+  // detected as non-tappable (they are tappable, just not while the viewport
+  // is at the current scroll position)
+  const reenableFixedAndStickyElementPointerEvents = disableFixedAndStickyElementPointerEvents();
+
+  let item;
+  while (item = enhancedTapTargets.shift()) {
+    const {tapTargetElement, largestRectCenterPoint, visibleClientRects} = item;
+
+    while (!pointIsInViewport(largestRectCenterPoint)) {
+      const previousScrollY = window.scrollY;
+      window.scrollTo(0, window.scrollY + window.innerHeight);
+
+      if (window.scrollY === previousScrollY) {
```

Comment:
I wonder how we should handle this case... it's not really supposed to happen so throwing might just be ok. On the other hand if we can't find it in the doc that means it's not visible, so we could also just set `isTop` to false

One thing we need to handle is scrollable divs that can contain elements outside the document  dimensions. Right now those are filtered out by `filterClientRectsWithinAncestorsVisibleScrollArea`, but it would be nice if could just get rid of all that logic.